transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -678,6 +682,7 @@ public void run() {
                     fireChannelInactiveAndDeregister(wasActive);
                 }
             }
+            afterClose();
```

Comment:
can you add a comment as to why this cannot be accomplished in `doClose()`? It seems strange that this operation is done synchronously and the `doClose()` call may use `closeExecutor`.